examples/RNOneSignal/App.js

```diff
@@ -8,484 +8,700 @@
+        console.log('isActive: ', openResult.notification.isAppInFocus);
+        console.log('openResult: ', openResult);
+
+        let debugMsg = 'OPENED: \n' + JSON.stringify(openResult.notification, null, 2);
+        this.setState({debugText:debugMsg}, () => {
+            console.log("Debug text successfully changed!");
+        });
+    }
+
+    /**
+     Once the user is registered/updated the onIds will send back the userId and pushToken
+        of the device
+     */
+    onIdsAvailable(device) {
+        console.log('Device info: ', device);
+        // Save the userId and pushToken for the device, important for updating the device
+        //  record using the SDK, and sending notifications
+        this.setState({
+            userId: device.userId,
+            pushToken: device.pushToken
+        });
+    }
+
+    /**
+     TODO: Needs to be implemented still in index.js and RNOneSignal.java
+     */
+    onSubscriptionChange(change) {
+        console.log('onSubscriptionChange: ', change);
+    }
+
+    /**
+     TODO: Needs to be implemented still in index.js and RNOneSignal.java
+     */
+    onPermissionChange(change) {
+        console.log('onPermissionChange: ', change);
+    }
+
+    /**
+     Success for the change of state for the email record? or setting subscription state of email record (so logging out)?
+     TODO: Validate functionality and make sure name is correct
+        Should match the onSubscriptionChange and
+
+     TODO: Validate this is working, might be broken after changing name
+     */
+    onEmailSubscriptionChange(change) {
+        console.log('onEmailSubscriptionChange: ', change);
+        this.setState({isEmailLoading:false});
+    }
+
+    /**
+     When an element on an IAM is clicked this will fire
+     The actionResult will contain information about the element clicked
+     */
+    onInAppMessageClicked(actionResult) {
+        console.log('actionResult: ', actionResult);
+
+        let debugMsg = 'CLICKED: \n' + JSON.stringify(actionResult, null, 2);
+        this.setState({debugText:debugMsg}, () => {
+            console.log("Debug text successfully changed!");
+        });
+    }
+
+    render() {
+        return (
+            <ScrollView style={styles.scrollView}>
+
+                { this.createTitleFields() }
+
+                <View style={styles.container}>
+
+                    { this.createSubscribeFields() }
+
+                    { this.createPrivacyConsentFields() }
+
+                    { this.createEmailFields() }
+
+                    { this.createExternalUserIdFields() }
+
+                </View>
+
+            </ScrollView>
+        );
+    }
+
+    /**
+     Create a red OneSignal Button with a name, loading state, and callback (onPress)
+     */
+    renderButtonView = (name, isLoading, callback) => {
+        let isPrivacyConsentButton = name.includes("Consent");
+
+        let isClickable = !isLoading
```

Comment:
nit, move this `isClickable` a few lines down, closer to the return below, so it is scope closer to where it is used.

Reply:
makes sense!